### PR TITLE
System chat messages

### PR DESCRIPTION
**Pull request by julianlam**
image

### CODE REVIEW

**Comment by julianlam** [comment]
Requires: https://github.com/NodeBB/nodebb-theme-persona/pull/485

File: src/messaging/create.js [hunk 1]
```diff
@@ -53,22 +53,24 @@ module.exports = function (Messaging) {
...
 		var message;
 		var isNewSet;
+		const timestamp = data.timestamp || new Date().getTime();
+		if (!callback) {
+			callback = function () {};
+		}
 
 		async.waterfall([
-			function (next) {
-				Messaging.checkContent(data.content, next);
```
Comment (barisusakli) [inline src/messaging/create.js]: Why is this removed? Maybe wrap it in `if (!data.system)` if you don't want system messages to be validated.

**Comment by julianlam** [comment]
Check commit messages 😀

File: src/messaging/create.js (hunk 1 above)
Comment (julianlam) [reply src/messaging/create.js]: Was initially removed because `.sendMessage()` contains the same call to check content. However, a data integrity check inside of `.addMessage()` is okay as well, as it might be called by plugins, etc.

I'll add it back in

**Comment by julianlam** [comment]
Hah oh right, the failing tests, knew I forgot something

**Comment by barisusakli** [comment]
Also this can be refactored into a function 
```
const message = await Messaging.addMessage({
		content: 'user-leave',
		uid: uid,
		roomId: roomId,
		system: 1,
});
Messaging.notifyUsersInRoom(uid, roomId, message);
```

Like so

async function addSystemMessage(content, uid, roomId) {
  const message = await Messaging.addMessage({
		content: content,
		uid: uid,
		roomId: roomId,
		system: 1,
  });
  Messaging.notifyUsersInRoom(uid, roomId, message);
}
```
Instead of duplicating code in 4 places

**Comment by julianlam** [comment]
Weird how the postgres tests failed :thinking:
